### PR TITLE
My pass at answering some questions

### PR DESCRIPTION
**Pull request by domenic**
- Some wording tweaks
- Added closebutton="" attribute
- Made attributes and properties generally more detailed
- Explained how showToast() connects to std-toast in more detail
- Added "actionclick" event (although it's still not super well fleshed out)
- Explained content model
- Removed duplicate toasts pattern; that shouldn't be common
- Added more TODOs for harder problems

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,64 +81,160 @@ which the standard toast aims to accomplish natively.
...
-### Attributes
+### The `<std-toast>` element
 
-*Note @me: view related goes in attributes*
+#### Attributes
 
-*TODO(me): how to structure `<std-toast>` info vs `showToast` info?*
+- [Global attributes](https://html.spec.whatwg.org/multipage/dom.html#global-attributes)
+- `open`: a boolean attribute, determining whether the toast is visible or not (according to the default styles). By default toasts are not shown.
+- `theme`: one of `"default"`, ???, or ???, conveying the semantic priority of the toast, and influencing its styling (both default and user-provided)
```
Comment (jackbsteinberg) [inline README.md:94]: We should maybe add some more about creating ones own theme (if that's something we want to support)

File: README.md [hunk 2]
```diff
@@ -75,64 +81,160 @@ which the standard toast aims to accomplish natively.
...
+- `position`: one of `"top-left"`, `"top-center"`, `"top-right"`, `"middle-left"`, `"middle-center"`, `"middle-right"`, `"bottom-left"`, `"bottom-center"`, or `"bottom-right"`. The default is (if the attribute is omitted or set to an invalid value) is `"bottom-right"`.
```
Comment (jackbsteinberg) [inline README.md]: There's an extra 'is' on this line, for some reason GitHub won't let me suggest a change

File: README.md [hunk 3]
```diff
@@ -75,64 +81,160 @@ which the standard toast aims to accomplish natively.
...
+    - TODO: this seems stylistic; should it be controlled via CSS instead?
+    - TODO: these are physical positions; should we use [logical ones](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Logical_Properties/Basic_concepts) instead?
+- `closebutton`: a boolean attribute, determining whether an explicit close button is shown. By default toasts do not have a close button.
```
Comment (jackbsteinberg) [inline README.md:98]: Why `closebutton` and not `closeButton`? Also, should we mention styling them / the default style?

File: README.md [hunk 4]
```diff
@@ -75,64 +81,160 @@ which the standard toast aims to accomplish natively.
...
+const toast = document.createElement('std-toast');
+console.log(toast.open); // false
+```
 
-*TODO(me): how to explain things that can be both attributes and properties (elt + js patterns)*
+#### Contents
 
-*TODO(me): where to put action and how to format it?*
+The `<std-toast>` element can be used as a generic container,
+but will work best (in terms of default styling and events) if the developer conforms to the following content model:
```
Comment (jackbsteinberg) [inline README.md:111]: What are our thoughts about icon and title? Should we add those as a pattern (e.g. any <h> element is styled as a title), leave it as a stretch goal, or just leave it out?

File: README.md [hunk 5]
```diff
@@ -75,64 +81,160 @@ which the standard toast aims to accomplish natively.
...
+```
 
-- `duration`
-- `closeButton`
-- `multiple` (??)
-- `newestOnTop` (??)
+These can equivalently be created via JavaScript:
+
+```js
+// TODO: write this once we figure out the JS API for supplying HTML messages,
```
Comment (jackbsteinberg) [inline README.md:144]: My current thinking is using template strings and just passing those in, but I'm still not sure what issues that extra complexity would add.

File: README.md [hunk 6]
```diff
@@ -75,64 +81,160 @@ which the standard toast aims to accomplish natively.
...
+Per container?
+How do we deal with different toasts having different values?
+
+#### Events
+
+A `<std-toast>` element can fire the following events:
+
+- `"open"`: the toast was shown
+- `"close"`: the toast was closed, either explicitly by the user, or via the timeout.
+  (Note: if animations were applied, the toast may not be entirely invisible at the time this event fires)
```
Comment (jackbsteinberg) [inline README.md:190]: Some toast libraries used "willClose" and "didClose" (as well as the same for open) to solve this visibility problem. I think it's worth looking into that again to see how popular that pattern was.

File: README.md [hunk 7]
```diff
@@ -75,64 +81,160 @@ which the standard toast aims to accomplish natively.
...
+`options` allows configuring both the attributes of the `<std-toast>`,
+and the options for this particular showing of the toast.
+Thus, the possible options are:
+
+- `theme`, like the attribute
+- `position`, like the attribute
+- `closeButton`, like the attribute
+- `duration`, like the `show()` option
+- `multiple`, like the `show()` option
+- `newestOnTop`, like the `show()` option
```
Comment (jackbsteinberg) [inline README.md:224]: After thinking about this I think you were right, we can probably capture `newestOnTop` in the multiple option, using values like `stackAbove` or `stackBelow`

File: README.md [hunk 8]
```diff
@@ -75,64 +81,160 @@ which the standard toast aims to accomplish natively.
...
+
+TODO: how do we deal with actions with this API? Ideas:
+- Provide an element
```
Comment (jackbsteinberg) [inline README.md:227]: / component for frameworks I suppose, but maybe element captures that sentiment

File: README.md (hunk 1 above)
Comment (domenic) [reply README.md:94]: Hmm. I feel like user-provided "theming" should be done via the closest-match theme plus `class=""`. `theme=""` is a bit more about the meaning of the toast, which does have an effect on the default styling, but isn't totally about that.

**Review by jackbsteinberg — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (domenic) [reply README.md:98]: All HTML attributes are lowercase. You can write them in any case (e.g. `` or `` or ``) but when documenting them we use lowercase.

File: README.md (hunk 4 above)
Comment (domenic) [reply README.md:111]: I'm really unsure. Maybe we should write in a TODO for now. I like the idea of supporting it as a pattern, but maybe as a stretch goal.

File: README.md (hunk 5 above)
Comment (domenic) [reply README.md:144]: Yeah, my worry there is that APIs which take raw HTML are a bit of a security issue. But maybe it'd be fine to start there...

File: README.md (hunk 6 above)
Comment (domenic) [reply README.md:190]: Sounds good. I will add a TODO for now. My worry is if there are custom animations we'd get thrown off.

File: README.md (hunk 8 above)
Comment (domenic) [reply README.md:227]: Yeah, the framework would convert their component to an element, when dealing with DOM APIs.

File: README.md (hunk 7 above)
Comment (domenic) [reply README.md:224]: I'll leave this as-is for now since I don't feel sure, but we can work on it as we prototype.

File: README.md (hunk 2 above)
Comment (jackbsteinberg) [reply README.md]:
```suggestion
- `position`: one of `"top-left"`, `"top-center"`, `"top-right"`, `"middle-left"`, `"middle-center"`, `"middle-right"`, `"bottom-left"`, `"bottom-center"`, or `"bottom-right"`. The default (if the attribute is omitted or set to an invalid value) is `"bottom-right"`.
```

File: README.md (hunk 3 above)
Comment (jackbsteinberg) [reply README.md:98]: Got it, good to know!

**Pull request merged**